src/context/QueryContext.h

```diff
@@ -124,6 +124,27 @@ class QueryContext {
         return symTable_.get();
     }
 
+    struct VarUser {
+        int64_t id;
+        bool    inLoop;
+    };
+
+    void addLastUser(const std::string &var, int64_t id) {
+        auto find = lastUser_.find(var);
+        if (find == lastUser_.end()) {
+            lastUser_.emplace(var, id);
+        }
+    }
+
+    StatusOr<int64_t> lastUser(const std::string &var) const {
+        auto find = lastUser_.find(var);
+        if (find == lastUser_.end()) {
+            return Status::Error("Can't find variable in user table.");
+        } else {
+            return find->second;
+        }
+    }
+
```

Comment:
I think we could put this part in symtable.

Reply:
Fixed.